torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -128,15 +128,41 @@ def _retrieve_or_adapt_input_to_graph_set(
                 List[int],
             ]
         ] = []
+        # onnx_tensor contains a list of scalars which could be one of
+        #   - tensor with empty shape,
+        #   - tensor with tensor with shape (1,),
+        #   - torch.SymInt,
+        #   - int
+        #   - ...
+        # They should all be promoted to tensor with shape (1,)
+        # in order to call ONNX's Concat.
         for tensor in onnx_tensor:
+            # Prepare `tensor` as input of ONNX's Concat.
+
             if isinstance(
                 tensor, torch.fx.Node
             ) and fx_type_utils.is_torch_symbolic_type(tensor.meta.get("val")):
-                sequence_mixed_elements.append(fx_name_to_onnxscript_value[tensor.name])
+                # In this case, tensor is a torch.SymInt from Dynamo's perspective.
+                # It might be mapped to tensor with shape () or (1,) in ONNX.
+                element_value = fx_name_to_onnxscript_value[tensor.name]
+                if isinstance(
+                    element_value, onnxscript_graph_building.TorchScriptTensor
+                ):
+                    # All elements sequence_mixed_elements will be send to onnx's Concat
```

Comment:
Could the assumption an ONNX Concat must happen be an early optimization? Why the concat is needed/expected here?

Reply:
SymInt will be a tensor computed by sub-graph in ONNX. No way to know the value in compilation time.